reactor-pool-micrometer/src/test/java/reactor/pool/introspection/micrometer/MicrometerTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2022 VMware Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.pool.introspection.micrometer;
+
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import org.junit.jupiter.api.Test;
+
+import reactor.core.publisher.Mono;
+import reactor.pool.InstrumentedPool;
+import reactor.pool.PoolBuilder;
+import reactor.pool.PoolMetricsRecorder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class MicrometerTest {
+
+	@Test
+	void metersRegisteredForGauge() {
+		SimpleMeterRegistry r = new SimpleMeterRegistry();
+		InstrumentedPool<String> pool = PoolBuilder.from(Mono.just("testResource")).buildPool();
+		InstrumentedPool.PoolMetrics poolMetrics = pool.metrics();
+
+		Micrometer.gaugesOf(poolMetrics, "testGauge", r);
+
+		assertThat(r.getMeters().stream()
```

Comment:
There is a `getMetersAsString` method which contains the name, the type, the tags and the values as well, might be simpler.

But even better, if you bring in `micrometer-test`, you can utilize [MeterRegistryAssert](https://github.com/micrometer-metrics/micrometer/blob/main/micrometer-test/src/main/java/io/micrometer/core/tck/MeterRegistryAssert.java) that lets you do things like: `assertThat(registry).hasMeterWithName("testGauge.resources.acquired")`, it also has options for tags.

Reply:
for the readability of the tests I was finding `getMeterAsString()` a bit too verbose. I've tried `micrometer-test`, and I don't find it much better in terms of asserting several things that we do assert easily here:

 - no unexpected meter is created (`containsExactly` part). there isn't even a way to assert the number of registered meters
 - for meters that are expected with a set of specific `KeyValues`, no way to assert the meter isn't registered with 0 tags (`assertThat(reg).hasMeterWithNameAndTagKeys("name")` matches "name" meters with any tag)